protocol/op-stack-manager-arch.md

```diff
@@ -0,0 +1,292 @@
+minProposalSizeBytes = 1
+challengePeriodSeconds = 1
+proofMaturityDelaySeconds = 1
+disputeGameFinalityDelaySeconds = 1
+```
+
+## Step 3: Deploy OP Chain Contracts
+
+We assume the latest Fault Proofs release `op-contracts/v1.5.0`.
+The exact set of sub-steps and inputs may change with future releases.
+
+The sub-steps are:
+
+1. Deploy `AddressManager`
+2. Deploy `ProxyAdmin` with it's owner set to `address(this)` and call `proxyAdmin.setAddressManager(addressManager)`
+3. Deploy `Proxy`s for:
+    1. `DisputeGameFactory`
+    2. `AnchorStateRegistry`
+    3. `DelayedWETH`
+    4. `PreimageOracle`
+    5. `MIPS`
+    6. `L1ERC721Bridge`
+    7. `OptimismPortal`
+    8. `SystemConfig`
+    9. `OptimismMintableERC20Factory`
+4. Deploy and configure the legacy proxied contracts and their implementations, `L1StandardBridge` and `L1CrossDomainMessenger`. (Details left out for brevity).
+5. Deploy the `FaultDisputeGame` and `PermissionedDisputeGame` contracts.
+6. Set and initialize all proxy implementations
+7. Transfer ownership of the `ProxyAdmin` to the input `proxyAdminOwner`.
+
+The L2 chain ID is an input needed to generate the salt for contract deployment.
+Other inputs for each sub-step are:
+
+1. None
+2. None
+3. None
+4. None
+5. Various FDG and PDG inputs: Left out for brevity
+6. Various contract inputs:
+    1. Roles: `proxyAdminOwner`, `systemConfigOwner`, `batcher`, `unsafeBlockSigner`, `proposer`, `challenger`
+    2. Fees: `basefeeScalar`, `blobBaseFeeScalar`
+    3. Fault Proof inputs: Left out for brevity
+7. None
+
+The outputs from each step are contract addresses.
+
+A sample TOML deploy config input for this step might look like:
+
+```toml
+l2ChainId = 10
+
+[roles]
+proxyAdminOwner = "0x123..."
+systemConfigOwner = "0x123..."
+batcher = "0x123..."
+unsafeBlockSigner = "0x123..."
+proposer = "0x123..."
+challenger = "0x123..."
+
+[fees]
+basefeeScalar = 1
+blobBaseFeeScalar = 1
+
+[fault_proofs]
+# left out for brevity
+```
+
+A sample TOML output for this step might look like:
+
+```toml
+systemConfigProxy = "0x123..."
+l1ERC721BridgeProxy = "0x123..."
+faultDisputeGame = "0x123..."
+addressManager = "0x123..."
+# etc.
+
+# Everything above this comment are the outputs, below are the inputs.
+l2ChainId = 10
+
+[roles]
+proxyAdminOwner = "0x123..."
+systemConfigOwner = "0x123..."
+batcher = "0x123..."
+unsafeBlockSigner = "0x123..."
+proposer = "0x123..."
+challenger = "0x123..."
+
+[fees]
+basefeeScalar = 1
+blobBaseFeeScalar = 1
+
+[fault_proofs]
+# left out for brevity
+```
+
+The initial version of these input files will be minimal and only include the required inputs to deploy standard chains.
+Future versions will include more inputs to support more complex deployments, such as custom gas tokens or alt-DA.
+The deploy scripts and OP Stack Manager code will be written in such a way to default to standard chains and only require additional inputs for non-standard chains.
+
+# Risks & Uncertainties
```

Comment:
To add:
- It needs to be somewhat extensible, to handle extra features like a challenge contract for Plasma
- We need a written summary of what current usage looks like, and figure out how we transition users from that. E.g. the need of a tool that converts a legacy `DeployConfig` into these smaller modular configs.

Reply:
These are great, added in 7a22832

Reply:
These are great, added in 7a22832